pkg/catalog/manager/traverse.go

```diff
@@ -316,6 +334,31 @@ func (m *Manager) traverseAndUpdate(helm *helmlib.Helm, commit string, cmt *Cata
 	return finalError
 }
 
+func (m *Manager) dropDeprecatedFields(cmt *CatalogInfo, catalogType string) error {
+	switch catalogType {
+	case client.CatalogType:
+		catalog := cmt.catalog
+		if catalog.Status.HelmVersionCommits == nil {
+			return nil
+		}
+		catalog.Status.HelmVersionCommits = nil
+	case client.ClusterCatalogType:
+		clusterCatalog := cmt.clusterCatalog
+		if clusterCatalog.Status.HelmVersionCommits == nil {
+			return nil
+		}
+		clusterCatalog.Status.HelmVersionCommits = nil
+	case client.ProjectCatalogType:
+		projectCatalog := cmt.projectCatalog
+		if projectCatalog.Status.HelmVersionCommits == nil {
+			return nil
+		}
+		projectCatalog.Status.HelmVersionCommits = nil
+	}
+	_, err := m.updateCatalogInfo(cmt, catalogType, "", true, true)
```

Comment:
Due to the way the function is written, this should be fine but I think if a function returns an updated version of an object you're passing in you should usually set it. In this case you may was to assign it to another variable and then do something like `*cmt = *cmt2`.

Reply:
When we set `updateOnly` to true, the catalog is eaten by the `updateCatalogInfo` method. This is also done multiple times throughout the traversal function. This is intentional, but I would normally agree if this was another circumstance.

Confirmed with @dramich